server/aws-lsp-codewhisperer/src/language-server/pathUtils.ts

```diff
@@ -0,0 +1,51 @@
+import os = require('os')
+
+/**
+ * Returns `true` if path `p` is a descendant of directory `d` (or if they are
+ * identical).
+ *
+ * Only the logical structure is checked; the paths are not checked for
+ * existence on the filesystem.
+ *
+ * @param d  Path to a directory.
+ * @param p  Path to file or directory to test.
+ */
+export function isInDirectory(d: string, p: string): boolean {
```

Comment:
I guess something like the following won't work:

```
function isInDirectory(d, p) {
    // Resolve paths to absolute paths
    const resolvedDir = path.resolve(d);
    const resolvedPath = path.resolve(p);

    // Get the relative path from the directory to the path
    const relativePath = path.relative(resolvedDir, resolvedPath);

    // Check if the relative path starts with `..` or is equal to ''
    return !relativePath.startsWith('..') && !path.isAbsolute(relativePath);
}
```

But what about using this: https://github.com/domenic/path-is-inside/blob/master/lib/path-is-inside.js